src/Altinn.Platform/Altinn.Platform.Storage/UnitTest/TestingControllers/InstancesControllerTests.cs

```diff
@@ -279,25 +279,157 @@ public async void Delete_ReponseIsDeny_ReturnsStatusForbidden()
                 Assert.Equal(HttpStatusCode.Forbidden, response.StatusCode);
             }
 
+            /// <summary>
+            /// Test case: Org user requests to get multiple instances from one of their apps.
+            /// Expected: List of instances is returned.
+            /// </summary>
+            [Fact]
+            public async void GetMany_OrgRequestsAllAppInstances_Ok()
+            {
+                // Arrange
+                string requestUri = $"{BasePath}?appId=ttd/complete-test";
+
+                HttpClient client = GetTestClient();
+                string token = PrincipalUtil.GetOrgToken("ttd", scope: "altinn:instances.read");
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+
+                int expectedNoInstances = 4;
+
+                // Act
+                HttpResponseMessage response = await client.GetAsync(requestUri);
+                string json = await response.Content.ReadAsStringAsync();
+                InstanceQueryResponse queryResponse = JsonConvert.DeserializeObject<InstanceQueryResponse>(json);
+
+                // Assert
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                Assert.Equal(expectedNoInstances, queryResponse.TotalHits);
+            }
+
+            /// <summary>
+            /// Test case: Org user requests to get all instances linked to their org.
+            /// Expected: List of instances is returned.
+            /// </summary>
+            [Fact]
+            public async void GetMany_OrgRequestsAllInstances_Ok()
+            {
+                // Arrange
+                string requestUri = $"{BasePath}?org=ttd";
+
+                HttpClient client = GetTestClient();
+                string token = PrincipalUtil.GetOrgToken("ttd", scope: "altinn:instances.read");
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+
+                int expectedNoInstances = 10;
+
+                // Act
+                HttpResponseMessage response = await client.GetAsync(requestUri);
+                string json = await response.Content.ReadAsStringAsync();
+                InstanceQueryResponse queryResponse = JsonConvert.DeserializeObject<InstanceQueryResponse>(json);
+
+                // Assert
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                Assert.Equal(expectedNoInstances, queryResponse.TotalHits);
+            }
+
+
+            /// <summary>
+            /// Test case: User requests to get multiple instances from a single instanceOwner - themselves.
+            /// Expected: List of instances is returned.
+            /// </summary>
+            [Fact]
+            public async void GetMany_PartyRequestsOwnInstances_Ok()
+            {
+                // Arrange
+                string requestUri = $"{BasePath}?instanceOwner.PartyId=1600";
```

Comment:
I would have liked to see that all this test had different userId and partyId values. This it to help ensure that we aren't mixing the two anywhere.

Reply:
onit :)